packages/framework/ai-collab/src/aiCollabApi.ts

```diff
@@ -0,0 +1,145 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { TreeNode, TreeView, ImplicitFieldSchema } from "@fluidframework/tree";
+// eslint-disable-next-line import/no-named-as-default
+import type OpenAI from "openai";
+
+/**
+ * OpenAI client options for the {@link AiCollabOptions} interface.
+ *
+ * @alpha
+ */
+export interface OpenAiClientOptions {
+	/**
+	 * The {@link OpenAI} client to use for the AI collaboration.
+	 */
+	client: OpenAI;
+	/**
+	 * The name of the target OpenAI model to use for the AI collaboration.
+	 */
+	modelName?: string;
+}
+
+/**
+ * Options for the AI collaboration.
+ *
+ * @alpha
+ */
+export interface AiCollabOptions<TSchema extends ImplicitFieldSchema> {
+	openAI: OpenAiClientOptions;
+	/**
+	 * The view of your Shared Tree.
+	 * @remarks Its is recommended to pass a branch of your current tree view so the AI has a separate canvas to work on
+	 * and merge said branch back to the main tree when the AI is done and the user accepts
+	 */
+	treeView: TreeView<TSchema>;
+	/**
+	 * The specific tree node you want the AI to collaborate on. Pass the root node of your tree if you intend
+	 * for the AI to work on the entire tree.
+	 * @remarks
+	 * - Optional root nodes are not supported
+	 * - Primitive root nodes are not supported
+	 */
+	treeNode: TreeNode;
```

Comment:
It isn't clear to me how these 2 parameters relate. Presumably the `treeNode` must belong to the specific `treeView` and be on the same branch. Given the tight coupling there, it would probably be better to take a single parameter that represents the (sub)tree being operated on (including its schema). We should have a type that captures this, though I honestly don't know what it is. @noencke @taylorsw04 any guidance on this?

Reply:
If the `treeView` parameter exists solely to support the `replaceRoot` operation, I would recommend removing that operation and the `treeView` param altogether for now.

I think it very unlikely that a developer would need to be able to replace the root of their application tree. And if they really do, they can adjust their app model to have an object at the root with an optional field beneath it that serves as the "root". Then they can just have the LLM edit that special root node like any other node in the tree.

Longer term, we should try to find ways to make the root of the shared tree less special, but for now I think it would simplify things a lot to remove support for direct operations on the root.

Reply:
Heh, I left a very similar comment on the signature to the editing method in agentEditReducer.ts. I think I'm more or less agree that at this point we should just remove the set root operation until we can figure out viable alternative. As I pointed out in my comment, this has downsides for trees that only have a single primitive in them. Unfortunately, I also think that the scoping concept as currently defined also has issues (consider the case where the scope node that is passed in is removed by the agent; currently, we likely crash when the scope is used in subsequent calls).

Reply:
My comment: https://github.com/microsoft/FluidFramework/pull/22836/files/b6755728ce844e69b98b9dd01ebb510111b187d0#r1828114781

Reply:
I'm going to test that removal case you called out and get back to you

Reply:
I have removed the setRoot operation as well as the ability to pass the TreeView in commit fb37d5500e19de5dd5a5664513592c997115e358, We can revisit adding it back later.